Patches.py

```diff
@@ -1846,13 +1906,20 @@ def update_scrub_text(message, text_replacement, default_price, price, item_name
     rom.write_int16(0xE2ADB6, 0x7057)
     buildAltarHints(world, messages, include_rewards=world.settings.misc_hints and not world.settings.enhance_map_compass, include_wincons=world.settings.misc_hints)
 
-    # Set Dungeon Reward actors in Jabu Jabu to be accurate
-    jabu_actor_type = world.get_location('Barinade').item.special['actor_type']
+    # Set Dungeon Reward Actor in Jabu Jabu to be accurate
+    # Vanilla and MQ Jabu Jabu addresses are the same for this object and actor
+    jabu_item = world.get_location(boss_map['Barinade']).item
+    jabu_stone_object = jabu_item.special['object_id']
+    rom.write_int16(0x277D068, jabu_stone_object)
+    rom.write_int16(0x277D168, jabu_stone_object)
+    jabu_stone_type = jabu_item.special['actor_type']
+    rom.write_byte(0x277D0BB, jabu_stone_type)
+    rom.write_byte(0x277D19B, jabu_stone_type)
+    jabu_actor_type = jabu_item.special['actor_type']
     set_jabu_stone_actors(rom, jabu_actor_type)
     # Also set the right object for the actor, since medallions and stones require different objects
     # MQ is handled separately, as we include both objects in the object list in mqu.json (Scene 2, Room 6)
     if not world.dungeon_mq['Jabu Jabus Belly']:
-        jabu_stone_object = world.get_location('Barinade').item.special['object_id']
         rom.write_int16(0x277D068, jabu_stone_object)
```

Comment:
This seems to be redundant with the code added above. Why is a change needed here anyway, besides inserting the `boss_map` indirection?

Reply:
This is a backport from Roman's fork.  Boss Shuffle began there and I ported it to Dev branch since I felt it more likely to be accepted here and then merged into Roman's that way.  I think I didn't realize that Dev didn't have it this way when rebasing.

Here is the original source: https://github.com/Roman971/OoT-Randomizer/blob/Dev-R/Patches.py#L1863..L1880

Reply:
The code added above was a leftover on my branch (I'm guessing due to some merge conflict), and I just removed it from Dev-R. I think you should be able to just adjust the existing code to use `boss_map` and omit the rest.

Here is the commit where I removed it for reference: <https://github.com/Roman971/OoT-Randomizer/commit/ba15f862a5cb67a0f1a8026b115d506df262cbf8>